### PR TITLE
Router: only render one page

### LINKED ISSUE #2075: Router in canary currently renders multiple Routes when they are matched
I've got two paths that are nearly identical:

1. `/o/{organizationId}/{dbConnectionId}/{exportConfigId}/edit`
2. `/o/{organizationId}/{dbConnectionId}/{exportConfigId}/{snapshotId}`

If I visit: `/o/1/2/3/edit`

It appears that both are being matched and rendered sequentially, but my inclination is that the first "edit" route should be the only one that's rendered.

### PR DESCRIPTION
**Pull request by Tobbe**
Fixes #2075

### CODE REVIEW

File: packages/router/src/__tests__/set.test.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ test('wraps components in other components', async () => {
   const TestSet = () => (
     <Router useAuth={window.__REDWOOD__USE_AUTH}>
       <Set wrap={[CustomWrapper, GlobalLayout]}>
-        <Route path="/" page={ChildA} name="childa" />
+        <ChildA />
```
Comment (peterp) [inline packages/router/src/__tests__/set.test.tsx:38]: I'm not sure I understand this change?

File: packages/router/src/RouteNameContext.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  routeName: string
+}
+
+const RouteNameContext = createNamedContext<RouteNameContextInterface>(
+  'RouteName'
+)
+
+export const RouteNameProvider = RouteNameContext.Provider
+
+export const useRouteName = () => {
```
Comment (peterp) [inline packages/router/src/RouteNameContext.tsx:15]: Nice!

File: packages/router/src/router.tsx [hunk 3]
```diff
@@ -251,9 +258,12 @@ const NotFoundChecker: React.FC<{ children: React.ReactNode }> = ({
         )
 
         if (match) {
+          // name will be undefined for redirect routes
+          routeName = name || ''
```
Comment (peterp) [inline packages/router/src/router.tsx:262]: Is there a case where we wouldn't have a routeName, and therefore `|| ''`?

**Review by peterp — APPROVED** [review]
Looks good to me, I left a few questions about things I'm not sure about - maybe they can be clarified with comments.

File: packages/router/src/router.tsx (hunk 3 above)
Comment (Tobbe) [reply packages/router/src/router.tsx:262]: Yes, isn't the reason clear from the comment?

Comment (peterp) [reply packages/router/src/router.tsx:262]: Ah, I was reading that as `undefined`, not an empty value!

Comment (peterp) [reply packages/router/src/router.tsx:262]: In other words... I was thinking `routeName` will be `undefined` for redirect routes, but then was left wondering why we made it an empty string.

Comment (peterp) [inline packages/router/src/router.tsx:262]: 🤷‍♂️ 
```suggestion
          routeName = name || '' // `name` is undefined for redirect routes
```

Comment (Tobbe) [reply packages/router/src/router.tsx:262]: You're right. I'll change the code a bit, try to make it easier to understand

File: packages/router/src/__tests__/set.test.tsx (hunk 1 above)
Comment (Tobbe) [reply packages/router/src/__tests__/set.test.tsx:38]: Now that we don't render multiple routes anymore, I changed it to just render the `<ChildA>` component directly. This is a test of the <Set> component, not the router :)

File: packages/router/src/router.tsx (hunk 3 above)
Comment (Tobbe) [reply packages/router/src/router.tsx]: I've changed this code, hopefully it's not as confusing anymore

**Review by peterp — APPROVED** [review]

File: packages/router/src/router.tsx (hunk 3 above)
Comment (peterp) [reply packages/router/src/router.tsx]: Nice!

Comment (Tobbe) [reply packages/router/src/router.tsx]: Discussion is continued in the other comment on the same line

**Comment by peterp** [comment]
I wonder what's up with the e2 tests.

**Pull request merged**